scipy/stats/_distn_infrastructure.py

```diff
@@ -2212,21 +2239,36 @@ def fit(self, data, *args, **kwds):
               function to be optimized) and ``disp=0`` to suppress
               output as keyword arguments.
 
+            - method : The method to use. The default is "MLE" (Maximum
+              Likelihood Estimate); "GMM" (Generalized Method of Moments)
+              is also available.
+
+            - order : The number of moments to fit with ``method="GMM"``.
+              This integer must be greater than or equal to the number of
+              non-fixed distribution parameters (including location and scale);
+              the default is the number of non-fixed parameters. This keyword
+              is ignored when ``method="MLE"``.
```

Comment:
2 questions here: 
1. What if the distribution has no finite (simple) moments? e.g. in the Cauchy case 
would an error be thrown?
2. Is it an optional fitting method when adding new distributions? e.g. new `rv_continuous` distributions should have an `MLE` but need not have a `GMM` method? 
If my understanding of the 2nd is correct then it may make sense to update the guide for adding new distributions with guidance.

Reply:
1. Good question. Currently, if calling the distributions `moment` method raises an exception, calling `fit` with `method="GMM"` will raise an exception - but we might want to edit the message. I'll be sure to test it.

2.  Hmm I noticed that some distributions override the `fit` method, but I didn't think about how this PR would affect that. 

>then it may make sense to update the guide for adding new distributions with guidance. 

I don't think overriding the `fit` method is addressed in the guide for adding new distributions yet, so sure, addressing it there is a good idea. (That doesn't belong in this PR, though, right?)  

> Is it an optional fitting method when adding new distributions?

Yes, I think so. When a contributor overrides `fit`, they have complete control. If they want to handle `method="GMM"` and `method="MLE"` differently, they can do that.  If the contributor of the new distribution chooses to override `fit` but doesn't want to have two separate methods, they should raise an `Exception` if the user calls for `method="GMM"` or warn somewhere that `method` will be ignored. 

Does that answer your question?

Reply:
>Does that answer your question?

I think so, I agree the addressing  in guide to adding new distributions need not be addressed in this PR, my comments are mostly call outs and questions, not necessarily a statement of how I think things should be done. 

On the overridden fit methods, I haven't thought much about that case, if things can remain unchanged that would be best I'd guess. It would show that `MME` is option, but some changes in overridden `fit`s may be necessary. I'd raise a not implemented exception on `MME` if and when necessary, unless you want to have a custom Exception. 
https://docs.python.org/3/library/exceptions.html#NotImplementedError

Reply:
1. When a non-finite value (`inf`, `nan`) is encountered during method of moments optimization, it raises an error.
```
ValueError: Method of moments encountered a non-finite distribution moment and cannot continue. Consider trying method='MLE'.
```
2. Often when fits are overridden, the documentation says something like:
```
This function uses explicit formulas for the maximum likelihood
        estimation of the exponential distribution parameters, so the
        `optimizer`, `method`, `loc` and `scale` keyword arguments are
        ignored.
```
I added `method` in all of these places.

Reply:
thanks for adding the `method` in the overridden `fit` functions. EDIT: ~Marking as resolved.~

on the first item when throwing a value error, can you include in the message which of the `k` moment(s) caused the error? 

If scipy is used in a distributed application, where someone doesn't have access to the env where the error was throw, these types of things go to logs, and it if there is more than a single moment it is difficult to determine the root cause of the issue, in these scenarios the more info (within reason) displayed in ValueError the better.